staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/kms/kms.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package kms transforms values for storage at rest using a KMS provider
+package kms
+
+import (
+	"crypto/aes"
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"sync"
+
+	"k8s.io/apiserver/pkg/storage/value"
+	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// defaultCacheSize is the number of decrypted DEKs which would be cached by the transformer.
+const defaultCacheSize = 1000
+
+// Factory allows creating various cloud KMS clients for implementing KEK-DEK based KMS encryption providers.
+type Factory interface {
+	NewGoogleKMSService(projectID, location, keyRing, cryptoKey string) (Service, error)
```

Comment:
I'm sorry, this factory shouldn't be here. It was removed long ago, but the rebase must not have picked it up. You can ignore this Factory interface completely.